Sources/NIO/EventLoop.swift

```diff
@@ -318,7 +318,7 @@ extension EventLoop {
     }
 
     /// Creates and returns a new `EventLoopPromise` that will be notified using this `EventLoop` as execution `Thread`.
-    public func newPromise<T>(file: StaticString = #file, line: UInt = #line) -> EventLoopPromise<T> {
+    public func newPromise<T>(type: T.Type = T.self, file: StaticString = #file, line: UInt = #line) -> EventLoopPromise<T> {
```

Comment:
I don't like the name `type` here. It sits weirdly with me: it's not the type of the promise, it's the type of what it contains. Not sure what the better name choice is though: `resultType`, maybe? Or `futureResult`? Or `producing`?

Reply:
@Lukasa `makePromise<T>(for type: T.Type = T.self)` ? that reads okay: 'make promise for integer'.

Reply:
well, in 1.0 obviously `newPromise`

Reply:
Works for me.

Reply:
@Lukasa updated

Reply:
Would `makePromise<T>(of type: T.Type = T.self)` be better?

Reply:
Seems I'm also late to the party here, and wanted to suggest `of` as well when I was reading the PR. "a promise of string (in the future)" somehow reads better than "a promise for string (in the future)". Your call tho

Reply:
we can still change it, it's not released. Let's page stdlib folk for API guidance, CC @airspeedswift / @lorentey

Reply:
ok, we now landed on `newPromise(of: T.Type = T.self)` because 'making a promise of a type' is more grammatical than 'making a promise for a type' -->  #672